composer/datasets/ffcv_utils.py

```diff
@@ -4,13 +4,50 @@
 import textwrap
 from typing import Optional
 
+import numpy as np
+
 from composer.core.types import Dataset
 
 log = logging.getLogger(__name__)
 
 __all__ = ["write_ffcv_dataset"]
 
 
+def _init_webdataset_meta_from_s3(remote: str) -> bytes:
+    """Read a WebDataset meta file from S3.
+
+    Args:
+        remote (str): S3 bucket or S3 bucket directory.
+    """
+    url = f'{remote}/meta.json'
+    cmd = 'aws', 's3', 'cp', url, '-'
+    ret = subprocess.run(cmd, capture_output=True)
+    assert not ret.stderr, 'Download failed, check your credentials?'
+    return ret.stdout
+
+
+def _init_webdataset_meta_from_local(remote: str) -> bytes:
+    """Read a WebDataset meta file from local filesystem.
+
+    Args:
+        remote (str): Local filesystem directory.
+    """
+    path = f'{remote}/meta.json'
+    return open(path, 'rb').read()
+
+
+def _init_webdataset_meta(remote: str) -> bytes:
```

Comment:
Thoughts on making _init_webdataset_meta in webdataset_utils non-underscored and reusing that?

If someone changes webdataset shard metadata fields in the future (eg, for non-perfectly-even shard sizes with no leftovers), this code could get overlooked and become broken

Reply:
Sounds good. Only issue I see is use of split in _init_webdataset_meta from webdataset_utils. In my case, split is already part of remote.

Reply:
I can change _init_webdataset_meta from webdataset_utils to make split optional and then reuse that. Does that work for you?

Reply:
I trust your excellent judgment, indeed that API is slightly awkward

Example prototyping some changes at https://github.com/mosaicml/composer/pull/846/files

Reply:
@knighton Fixed it. Let me know if it looks good now.